.travis.yml

```diff
@@ -121,331 +147,91 @@ matrix:
-        # Remove bad openjdk6 from trusty image, so
-        # Pants will pick up oraclejdk6 from `packages` above.
-        - sudo rm -rf /usr/lib/jvm/java-1.6.0-openjdk-amd64
-        - sudo rm -rf /usr/lib/jvm/java-6-openjdk-amd64
-        - jdk_switcher use oraclejdk8
+    # Integration tests - shard 1
+    - <<: *default_test_config
       env:
         - SHARD="Python integration tests for pants - shard 1"
       script:
         - ./build-support/bin/ci.sh -x -efkmrjlpnt -i 0/7 "${SHARD}"
 
-    - os: linux
-      dist: trusty
-      sudo: required
-      addons:
-        apt:
-          packages:
-            - lib32stdc++6
-            - lib32z1
-            - lib32z1-dev
-            - gcc-multilib
-            - python-dev
-            - openssl
-            - libssl-dev
-      stage: Test Pants
-      language: python
-      python: "2.7.13"
-      before_install:
-        # Remove bad openjdk6 from trusty image, so
-        # Pants will pick up oraclejdk6 from `packages` above.
-        - sudo rm -rf /usr/lib/jvm/java-1.6.0-openjdk-amd64
-        - sudo rm -rf /usr/lib/jvm/java-6-openjdk-amd64
-        - jdk_switcher use oraclejdk8
+    # Integration tests - shard 2
+    - <<: *default_test_config
       env:
         - SHARD="Python integration tests for pants - shard 2"
       script:
         - ./build-support/bin/ci.sh -x -efkmrjlpnt -i 1/7 "${SHARD}"
 
-    - os: linux
-      dist: trusty
-      sudo: required
-      addons:
-        apt:
-          packages:
-            - lib32stdc++6
-            - lib32z1
-            - lib32z1-dev
-            - gcc-multilib
-            - python-dev
-            - openssl
-            - libssl-dev
-      stage: Test Pants
-      language: python
-      python: "2.7.13"
-      before_install:
-        # Remove bad openjdk6 from trusty image, so
-        # Pants will pick up oraclejdk6 from `packages` above.
-        - sudo rm -rf /usr/lib/jvm/java-1.6.0-openjdk-amd64
-        - sudo rm -rf /usr/lib/jvm/java-6-openjdk-amd64
-        - jdk_switcher use oraclejdk8
+    # Integration tests - shard 3
+    - <<: *default_test_config
       env:
         - SHARD="Python integration tests for pants - shard 3"
       script:
         - ./build-support/bin/ci.sh -x -efkmrjlpnt -i 2/7 "${SHARD}"
 
-    - os: linux
-      dist: trusty
-      sudo: required
-      addons:
-        apt:
-          packages:
-            - lib32stdc++6
-            - lib32z1
-            - lib32z1-dev
-            - gcc-multilib
-            - python-dev
-            - openssl
-            - libssl-dev
-      stage: Test Pants
-      language: python
-      python: "2.7.13"
-      before_install:
-        # Remove bad openjdk6 from trusty image, so
-        # Pants will pick up oraclejdk6 from `packages` above.
-        - sudo rm -rf /usr/lib/jvm/java-1.6.0-openjdk-amd64
-        - sudo rm -rf /usr/lib/jvm/java-6-openjdk-amd64
-        - jdk_switcher use oraclejdk8
+    # Integration tests - shard 4
```

Comment:
The comments don't seem to help much at all on these abbreviated shards since `env` is very nearby. On the handful of remaining shards its maybe marginally hard? Seems redundant to me anyway.